bin/webpack

```diff
@@ -7,7 +7,7 @@ require "yaml"
 ENV["RAILS_ENV"] ||= "development"
 RAILS_ENV = ENV["RAILS_ENV"]
 
-ENV["NODE_ENV"] ||= RAILS_ENV
+ENV["NODE_ENV"] ||= RAILS_ENV == "integration" ? "production" : RAILS_ENV
```

Comment:
so, never ever using `config/webpack/integration.js` ?

Reply:
Heh, that would explain a lot.  Had it in my head that it was based on `RAILS_ENV` for some reason.

Might not be needed at this point now.  Good catch.

Reply:
So for this, either I am going to have to change this line to also consider the `CYPRESS_DEV` environment variable, or update the `WEBPACK_CONFIG` to be based off the `RAILS_ENV`.  I will mess around with both.

Reply:
So still having issues with this one:

Currrently trying to get this to work using the `cypress:ui:dev:*` tasks, but currently I am not have any luck and it is still serving assets as if `rake compile_assets` was run...

This is what the first request looks like to `/` with the `cypress:ui:dev:*` tasks currently:

<details>
<summary>With <code>CYPRESS_DEV=1</code>...</summary>

```
yarn cypress open
yarn run v1.22.4
$ ./manageiq-ui-classic/node_modules/.bin/cypress open
GET /__/ 200 5.160 ms - -
GET /__cypress/runner/cypress_runner.css 200 4.320 ms - -
GET /__cypress/runner/cypress_runner.js 200 2.963 ms - -
GET /__cypress/static/favicon.ico 200 1.726 ms - -
GET /__cypress/runner/fonts/fa-solid-900.woff2 200 0.961 ms - 76120
GET /__cypress/iframes/integration/ui/login.spec.js 200 2.914 ms - 928
GET /__cypress/tests?p=cypress/integration/ui/login.spec.js 200 853.201 ms - -
GET /__cypress/tests?p=cypress/support/index.js 200 988.671 ms - -
GET / 200 21.848 ms - -
GET /assets/application-8c1b6540231eb382fedf6eaa2a9d341be9bf5fd7f5a8967107925b7bcada2e36.css 200 34.889 ms - -
GET /custom.css 200 55.549 ms - -
GET /packs/runtime-c8499fefc3eae562558a.js 200 71.612 ms - -
GET /packs/manageiq-ui-classic/toolbar-actions-common-16402a56f4a5b765479c.js 200 39.832 ms - -
GET /packs/manageiq-ui-classic/provider-dialogs-common-dec78427eff0d58f38b6.js 200 30.099 ms - -
GET /packs/manageiq-ui-classic/globals-0713c97bc82c18c03379.js 200 10743.570 ms - -
GET /packs/manageiq-ui-classic/provisioning-actions-common-1fc10b00ecfa6585615d.js 200 10727.937 ms - -
GET /packs/manageiq-providers-lenovo/component-definitions-common-263d8a8005e0d4d25223.js 200 39.213 ms - -
GET /packs/manageiq-providers-nuage/component-definitions-common-ca09234f868df5fc6b89.js 200 10718.134 ms - -
GET /packs/manageiq-ui-classic/angular-common-8824e9177498da64d45c.js 200 10733.576 ms - -
GET /packs/manageiq-providers-redfish/component-definition-common-53b9b1ea8b82afff6dfc.js 200 10717.945 ms - -
GET /packs/manageiq-providers-nsxt/component-definitions-common-04b2ddc2d48b741bb40a.js 200 10717.909 ms - -
GET /packs/manageiq-ui-classic/application-common-b9bf17a5fb9cbec243c5.js 200 10738.514 ms - -
GET /packs/manageiq-ui-classic/component-definitions-common-dd79b55792eee324d837.js 200 10728.216 ms - -
GET /packs/shims-276f94d5665d34c48faa.js 200 70.678 ms - -
GET /assets/application-fb7fc51a35d47efda08e8eca285b3157fe8892d91360d69acbb57892aa1cd4e3.js 200 10830.817 ms - -
GET /packs/vendor-870bd1a3312bc7b488b5.js 200 70.735 ms - -
GET /assets/layout/login-screen-logo-cd43380036fc96964823fd8d6d7486fe9bcfcce1498daf0c41d8bc94385511da.png 200 6.904 ms - -
GET /assets/layout/brand-238fd8d5ec6a8907af00d80292c5a2a1d2becb340930ed421d71605e022c2193.svg 200 6.320 ms - -
GET /assets/layout/bg-login-7ca4aca7fb1ab1a1ea4eb3455c83934c07906172f4ea784c5ef6af311a5f3171.png 200 3.451 ms - -
GET /assets/OpenSans-Regular-webfont-3cfb28778895d6adca324710b2000c6e15ef5a7b88d461f39b29ff6fb877b778.woff2 200 4.295 ms - -
GET /assets/OpenSans-Italic-webfont-20e0080dce3846c93b80cc230cbd1de81b6b8ddd37f58ae69faa28f46b2925aa.woff2 200 4.206 ms - -
GET /assets/OpenSans-Semibold-webfont-dabab8829a3f91d63ed1f8430d439c5dad1d35aca216e69cd842b58d2dac4f9c.woff2 200 3.227 ms - -
GET /assets/layout/overlay_dark-f5527015a0bcd6be16704dc4973a6b6955e5e7ccced936f9926eac2655182b4b.png 200 3.869 ms - -
```

</details>

Ideally they would look like the following in development mode:

<details>
<summary>User <code>DB=vmdb_integration bin/rails c</code> (assumes <code>RAILS_ENV=development</code>)...</summary>

```
yarn cypress open
yarn run v1.22.4
$ ./manageiq-ui-classic/node_modules/.bin/cypress open
GET /__/ 200 5.101 ms - -
GET /__cypress/runner/cypress_runner.css 200 24.336 ms - -
GET /__cypress/runner/cypress_runner.js 200 1.660 ms - -
GET /__cypress/static/favicon.ico 200 1.035 ms - -
GET /__cypress/runner/fonts/fa-solid-900.woff2 200 0.805 ms - 76120
GET /__cypress/iframes/integration/ui/login.spec.js 200 2.552 ms - 928
GET /__cypress/tests?p=cypress/integration/ui/login.spec.js 200 699.547 ms - -
GET /__cypress/tests?p=cypress/support/index.js 200 839.903 ms - -
GET /__cypress/runner/fonts/fa-regular-400.woff2 200 1.697 ms - 13600
GET / 200 18.900 ms - -
GET /packs/runtime-c8499fefc3eae562558a.js 200 51.303 ms - -
GET /custom.css 200 107.622 ms - -
GET /packs/manageiq-ui-classic/globals-0713c97bc82c18c03379.js 200 268.350 ms - -
GET /packs/shims-276f94d5665d34c48faa.js 200 41.989 ms - -
GET /assets/application.self-3e5ec5b996d3ebb38e1f8d5c5cb1124ddb634e7bb48bf63bf92ca24ef891f6cb.css?body=1 200 554.235 ms - -
GET /assets/jquery_overrides.self-69a90f438969e9edd017fd5b4ab92452f32ce4b7d175d6da20088804dd6324c3.js?body=1 200 472.802 ms - -
GET /assets/locale/en/app.self-e4bc104d3049fd7fa3e9ef22e826e6fcca2d7b6fc56628b3f12cc61b93c10d55.js?body=1 200 540.469 ms - -
GET /assets/controllers/container_dashboard/container_project_dashboard_controller.self-b96f7b77ee23cea9b712bfaabe89941fb5aa4973f436efbf04f0f72bd9f27a4d.js?body=1 200 451.538 ms - -
GET /assets/controllers/container_dashboard/util.self-fc711fbc25a6150b95c616df1503669c8ac9d0e4d58ec98fb047cf6617c391fc.js?body=1 200 392.252 ms - -
GET /assets/controllers/container_dashboard/util/dashboard-utils-factory.self-cb9fe325ec6e5b4fea588b9af69c4a19503de5c0a6220394a2b86b923f20fea1.js?body=1 200 391.347 ms - -
GET /assets/controllers/credentials/credentials_controller.self-89038a1bb757da235e4a26b82d4d19b4a6f948c6b7c95b7a86842a2b6acf1d6d.js?body=1 200 384.942 ms - -
GET /assets/controllers/dialog_editor/dialog_editor_controller.self-eab480930b40631aaf076a2541a814d43c146f5dd5cfb724ef5033e22a3f9739.js?body=1 200 393.818 ms - -
GET /assets/controllers/dialog_user/dialog_user_controller.self-1e48c3fbab03ff55bb1dfdc53c15ccd03f377697a9b5387fc728ec95487c0c58.js?body=1 200 400.628 ms - -
GET /assets/controllers/dialog_user/dialog_user_reconfigure_controller.self-a05cdbd1652326324850c086715ad463f4244c5604cfe7ba61cf73fe50982cbb.js?body=1 200 402.712 ms - -
GET /assets/controllers/ems_common/aggregate_status_card_controller.self-efa3842a7338d620e5b041bba5f34529b772b31743256ab717be9682a1354fc8.js?body=1 200 400.136 ms - -
GET /assets/controllers/ems_common/ems_common_form_controller.self-290ed64c74a843ce04edf1d5f3d089849269f1e6e25a5bb9ffe51d581de97232.js?body=1 200 364.762 ms - -
GET /assets/controllers/ems_container_dashboard/aggregate_status_card_controller.self-01a4e644b1bbcd1e3b199c7526880524af4944bd02e4144854d7e1ff2c912bae.js?body=1 200 385.256 ms - -
GET /assets/controllers/ems_container_dashboard/heatmap_controller.self-a0c209571932d13a674c647627f4f8d2d1d313ef1ce7893ce95f0f9f35ed42fc.js?body=1 200 384.471 ms - -
GET /assets/controllers/ems_container_dashboard/recent_images_line_chart_controller.self-91c0d080a2a3ff9abacf436191f2b2d98089e2b4c0db718a82bd377b27a2ea8c.js?body=1 200 393.631 ms - -
GET /assets/controllers/ems_container_dashboard/recent_pods_line_chart_controller.self-e53900da53a4e333d7f4c67d96d399f675e0ea1ae5b55cb9ba4c1aa862c108b9.js?body=1 200 397.759 ms - -
GET /assets/controllers/ems_container_dashboard/utilization_trend_chart_controller.self-bb0d911576da60ed7361990ff73a9cd46fcae55fc3a24e62f9706bb08c7bc87d.js?body=1 200 794.786 ms - -
GET /assets/controllers/ems_container_dashboard/trends_chart_controller.self-cbe2cf148f5fb49f833e4d4276fe47af3a1619da0d62e00f84da997c9cbf6131.js?body=1 200 809.808 ms - -
GET /assets/controllers/ems_infra_dashboard/heatmap_controller.self-dc8726b754eddbada0c205a9590bd683d7c558b97ebf4d94edfe3060cbb8f874.js?body=1 200 832.255 ms - -
GET /assets/controllers/ems_infra_dashboard/utilization_trend_chart_controller.self-f7029607546c017195d223555f336db737c8d89caab6fe0f7c74e910643e8fbf.js?body=1 200 818.138 ms - -
GET /assets/controllers/ems_keypair/ems_keypair_controller.self-d665293232fdd7202df9c6f8b428a84f7e5f7b1d33ff8e74749628986efa2ef6.js?body=1 200 409.762 ms - -
GET /assets/controllers/ems_physical_infra_dashboard/recent_servers_line_chart_controller.self-c8a51d1077222cbab7421a0c9f18789b0f2ae3adc7a03622480e366af2af0443.js?body=1 200 429.932 ms - -
GET /assets/controllers/ems_physical_infra_dashboard/server_health_pie_chart_controller.self-c7ec1219d419a5a57d102f4968a61e52dca200e81e94efcacd16fe04c58de069.js?body=1 200 418.642 ms - -
GET /assets/controllers/error_modal_controller.self-fc900eb2dcfae8abf93da89e98f71bdc92730fe44c1d0918298984a311de8758.js?body=1 200 433.519 ms - -
GET /assets/controllers/floating_ip/floating_ip_form_controller.self-c16c56b9fa329f2c1be3d05e287e0a715a6f28c3b851444945d458e77248da29.js?body=1 200 610.120 ms - -
GET /assets/controllers/fonticon_picker_controller.self-9381f77b3669c7057550cb8862ea7b110a2910ac4d36996a98f1a00a1bbc7bc8.js?body=1 200 591.518 ms - -
GET /assets/controllers/host/host_form_controller.self-35d7ed8a9c75461cd9c3f93f92f2c82f64ab7c70d4df44e9f527e10d5511d4e9.js?body=1 200 601.598 ms - -
GET /assets/controllers/host_aggregate/host_aggregate_form_controller.self-8d890776aa87c765c968228fd1be169d67646e957e44e9eb0694f7913accb32a.js?body=1 200 583.293 ms - -
GET /assets/controllers/live_metrics/ad_hoc_metrics_controller.self-f07ff4deb431a293e026c34fbcb1af3ec924d1762f4e0200727da15544acad41.js?body=1 200 396.617 ms - -
GET /assets/controllers/live_metrics/util/metrics-config-factory.self-9776f90847d878706ea922c3668ed9286fce3b91f97d9be56c7eb38650b4107c.js?body=1 200 402.821 ms - -
GET /packs/vendor-870bd1a3312bc7b488b5.js 200 47.498 ms - -
GET /assets/controllers/live_metrics/util/metrics-http-factory.self-5d5bd62edfa8c40810294149214a32e698680ee314372a5d5a2843e62fadb976.js?body=1 200 391.326 ms - -
GET /assets/controllers/live_metrics/util/metrics-parse-url-factroy.self-54425966dbbdb18068cd4c0d60f42c48cec78561746648a86071d28394e9a833.js?body=1 200 396.569 ms - -
GET /assets/controllers/live_metrics/util/metrics-utils-factory.self-e6c0e1a5ab7d500d3b72f37dc658629f275cdeb725917d760221402d59d5aca8.js?body=1 200 389.271 ms - -
GET /assets/controllers/miq_ae_class/ae_inline_method_selection_controller.self-049fb1ed2302e78df7a44b6fb0b60a9072cd8e03d9c2490338263948037a021a.js?body=1 200 393.550 ms - -
GET /assets/controllers/mixins/charts-mixin.self-38670771dea7a32e476a1ed4f5deae32ab9da9e1ee64ba2ff752a1349ee6ca77.js?body=1 200 390.910 ms - -
GET /assets/controllers/network_router/network_router_form_controller.self-9358e427d7003a229fc95dcac538ec5fd5654538a46fbd26760a714683cf0e3e.js?body=1 200 392.558 ms - -
GET /assets/controllers/ops/diagnostics_database_form_controller.self-3733cadf2aabd9a86a5e212c7aba7d7e703baf5379544f3b5e348506237c4c7b.js?body=1 200 395.494 ms - -
GET /assets/controllers/container_dashboard/card/heatmaps/heatmaps-card-directive.self-a19eca9244decb2715ecfad8fd0cc3460ec79549e418088be525c553f2b01319.js?body=1 200 4490.207 ms - -
GET /assets/controllers/ops/pglogical_replication_form_controller.self-1dc695ac34d303df097e50044cf5529c7069b83f196456ee1a3f19e3faeb3794.js?body=1 200 408.989 ms - -
GET /assets/controllers/orchestration_template/orchestration_template_copy_controller.self-821a25d4fd054410443aa0e9f5bf3459f9177058a567705d8eea93660b122e96.js?body=1 200 396.056 ms - -
GET /assets/controllers/ops/tenant_quota_form_controller.self-63d202faa9a41294f61c19f8c0b4dc7cc7c07c8f86406b8035f864c9436dd0f7.js?body=1 200 516.419 ms - -
GET /assets/controllers/playbook-reusable-code-mixin.self-dd34cdfd06348a3a1d2efb68232c0e9b25346a456aea502149895cc83d64ae9a.js?body=1 200 460.193 ms - -
GET /assets/controllers/report_data_controller.self-85b2263e6da164cfb373618627ac0ddc434ac3f657d626913f74ef3d6ea2709c.js?body=1 200 435.702 ms - -
GET /assets/controllers/schedule/schedule_form_controller.self-d850caeef3a38a5b6f42f72044bc6b031e47caee4a58d1e9b6525468a8e65c3c.js?body=1 200 435.384 ms - -
GET /assets/controllers/security_group/security_group_form_controller.self-7fd39cdd5a421d5bcc5f98bca330d8e151f7e5101daeaa37353551b8c3db4118.js?body=1 200 491.681 ms - -
GET /assets/controllers/timeline/timeline_options_controller.self-4fecb11f6536ee5f1ecbdbfc8f38295513f135579fcc12fb6caf8630e13613a5.js?body=1 200 512.016 ms - -
GET /assets/controllers/topology/cloud_topology_controller.self-fa4114613a0680a138c85581b9967b66eba75b5811c174d81ccd6f24d4174f53.js?body=1 200 497.157 ms - -
GET /assets/controllers/topology/container_topology_controller.self-032febac411b25daea96c5064bd59dcb41010878d111c5d31cd318b0f366a105.js?body=1 200 473.070 ms - -
GET /assets/controllers/topology/infra_topology_controller.self-140e878ba4a5326101953aee3539b84caef22104c8b7bf9bf65c5119217ab858.js?body=1 200 471.039 ms - -
GET /assets/controllers/topology/network_topology_controller.self-f92381305ccc12fb27b2fa4baa5846d4a3c1eeb8c05b54be8c8bfd7531b002f5.js?body=1 200 401.233 ms - -
GET /assets/controllers/topology/physical_infra_topology_controller.self-6283f5f4e90511fcfe757c1a97b3326ec180689c1375a0016c3dbeb292e79a52.js?body=1 200 400.692 ms - -
GET /assets/controllers/tree_view_controller.self-4683d222a4e1c4fb365906d4b2aa25b7f8278323636f158f6ce235f662835545.js?body=1 200 394.648 ms - -
GET /assets/controllers/vm_cloud/vm_cloud_associate_floating_ip_form_controller.self-726f932d4f921eadebec305d09bd3294365e5fe0ab81cac2d0e289828d752c2a.js?body=1 200 367.673 ms - -
GET /assets/controllers/vm_cloud/vm_cloud_attach_form_controller.self-3e720a7cf52bc61a8584beda5aa1d2257c185b77a901d910e6b85c8d214718d7.js?body=1 200 368.129 ms - -
GET /assets/controllers/vm_cloud/vm_cloud_detach_form_controller.self-952e1acefaec4dccac436266b8505864bbba39ce97f846eb1aa1797487cf0330.js?body=1 200 372.614 ms - -
GET /assets/controllers/vm_cloud/vm_cloud_disassociate_floating_ip_form_controller.self-cf7f89ec0acef238078c872101f5323feb09fbc2e05ee1e7c138648cca08e31b.js?body=1 200 363.292 ms - -
GET /assets/controllers/vm_cloud/vm_cloud_live_migrate_form_controller.self-8a9aa67d741f55dc96186c5f123f2d36ef5fcf8896d35a31a4dfd56164d3202a.js?body=1 200 358.251 ms - -
GET /assets/controllers/vm_cloud/vm_cloud_resize_form_controller.self-eb8da4f8126d5293e53faf3e8018629bbfdbdcd5dd5dd3eee316f2e771b1b5ab.js?body=1 200 366.806 ms - -
GET /assets/directives/ansible_credentials/adjust_on_reset.self-eb7248ad1ee24acf635888ee458ecd1e5e9daa4ee5974fe86aeb11818958c2da.js?body=1 200 356.488 ms - -
GET /assets/directives/autofocus.self-10d9a419d87d62f2ba60e680c38f940ba8a90019b76d3a33335edaa65b8105cc.js?body=1 200 343.871 ms - -
GET /assets/directives/checkchange.self-ca6d96718416774401c5b2ae841e5fea622d588ecf3d5e30eda27fd92ce4203f.js?body=1 200 345.151 ms - -
GET /assets/directives/clear_field_set_focus.self-9648650cf0c5001a38487ebf8e07aec7ef35001996c1ed2b8153abbe4ced95f3.js?body=1 200 348.041 ms - -
GET /assets/directives/configuration/all_time_check.self-6ae4a3f2d7e327cb2c221530240f0d6581fe34618eae836f32794299d0bc4c7c.js?body=1 200 338.460 ms - -
GET /assets/directives/configuration/min_time_check.self-aaf00168ef09ede5a6023227bf5d9d3e87443ae2506bd58555091880df185cc8.js?body=1 200 342.299 ms - -
GET /assets/directives/configuration/some_time_check.self-599c4d44233ca419cbe18a1d45f6e78a9902bb28dcf86cd2ea9c2784c1a4b043.js?body=1 200 337.516 ms - -
GET /assets/directives/datetimepicker.self-e4d58ebdd5a6f0e00b036bba52e299622f89cde58866bb25289f99d2e1222634.js?body=1 200 337.040 ms - -
GET /assets/directives/detect_spaces.self-839360e044680302cac20497856f376619f849b999082319c494bfccc9c20b2b.js?body=1 200 333.457 ms - -
GET /assets/directives/form_changed.self-5792fb8c0533013064473e065e3a5e503a7f75aa4853497b08f50a31a597aa7c.js?body=1 200 331.537 ms - -
GET /assets/directives/greater_or_zero.self-20a15bacee2011ad1dee41dda192004d922eaf2d8eff55d67b34d8fdde1c5497.js?body=1 200 330.902 ms - -
GET /assets/directives/greater_than_zero.self-56cebc38ea58a1d0977ce9cb5d5db2a08734fd029fe93d568c1d245e3201bebf.js?body=1 200 338.582 ms - -
GET /assets/directives/hostname_validation.self-0b983fe55b1399e3acd73acf220220ec62a793458ed13187a2c8cab0829873dc.js?body=1 200 340.301 ms - -
GET /assets/directives/miq_datepicker.self-4206bd8660a446e642c1ae28c85083244902304d104b1e36e687c89640d1316b.js?body=1 200 343.851 ms - -
GET /assets/directives/miq_form.self-dd2d4caca5f2cd0eaa61842c56c5876273f77b957f002266027d6bfa32656b06.js?body=1 200 346.894 ms - -
GET /assets/directives/miqrequired.self-f56283fd5070c4c74c83af008671c3635d15ed3ac0ef63fd80c82e9b04b786df.js?body=1 200 345.834 ms - -
GET /assets/directives/reconfigure/validateMultiple.self-e7ed7622cd30239ad5f5d0bdf3ebc531c587e1212a036bf6f154845027c506b1.js?body=1 200 351.033 ms - -
GET /assets/directives/reconfigure/validateTotal.self-1cf0067a76dbda351373cd04d17374cc79a63eea4ec08212e95b6a3bdcc01098.js?body=1 200 339.177 ms - -
GET /assets/directives/require_file.self-f0e57f9ab4c514c8ddf076b3d0efb34b170d19a289dc80ef4ad2780c9f9f8735.js?body=1 200 340.481 ms - -
GET /assets/directives/required_depends_on.self-f5da63632422e101f8c327e9944b9ef6918a7a3fd5585b3197d9916931eaaa22.js?body=1 200 336.131 ms - -
GET /assets/directives/scheduler/updateDropdownForFilter.self-47cbef952c71acdd602c1314ebe2ee8f388db79cfb89f5d568126f73ea401ea5.js?body=1 200 338.321 ms - -
GET /assets/controllers/reconfigure/reconfigure_form_controller.self-bfb3101bc05938ba1ccda856af6f6d2be507b77e5c8c516406415374f98dc5b1.js?body=1 200 3289.561 ms - -
GET /assets/directives/selectpickerForSelectTag.self-918b32502975f70fc6ec8a18b5f8624d8ececbbacbfef03c5e386f59e836c235.js?body=1 200 340.833 ms - -
GET /assets/directives/translate.self-581a268c812f8c8673b37377f758461508074fe41e9d152d04fd25ae1d0ea84e.js?body=1 200 340.337 ms - -
GET /assets/directives/url_validation.self-dfc6d89606bd2dabff8b7aa79f753a337a56623a89f4ec51fdb6022a5d50e7b7.js?body=1 200 428.996 ms - -
GET /assets/directives/validation/adjust_error_on_tab.self-2949274cac2585a76b949a53d8b3592d0493b49c77f0bbf6ee0506b3ff045888.js?body=1 200 432.018 ms - -
GET /assets/directives/validation/error_on_tab.self-68e26a59f0fb75e197f3d97ad81cee88dd5177e9405e2ae3fbe3892de801fbf8.js?body=1 200 431.378 ms - -
GET /assets/directives/validation/reset_validation_status.self-e3d818a993a155498cdabf869f276301c86ee87c2035f9503d29c2082b6a107a.js?body=1 200 429.368 ms - -
GET /assets/directives/validation/validation_status.self-9fe7947603d5fce69d1b6e756366907e139bb6a35181a51c6823fba2f5d7d1d7.js?body=1 200 425.198 ms - -
GET /assets/directives/scheduler/updateDropdownForTimer.self-e5a54a2a7c8af10f7d36722f0686616fb23943c5dec36eee1a6fd72d4d45b0d4.js?body=1 200 1027.884 ms - -
GET /assets/components/ae_resolve_options/ae-resolve-options.self-4215cc2a2d85704665ce04e12854973e40a93899770e1d0152a7c9e6c236e1d5.js?body=1 200 341.234 ms - -
GET /assets/components/ansible-raw-stdout.self-a230d6222653a3a5f21a6ea2e7fdaa9ccc738bc00d8266b9a5338532bddc0cf1.js?body=1 200 333.101 ms - -
GET /assets/components/ansible-credential-options.self-54f4a61c99e38bc2be9b11913cb749acc1421cd44af01cfbde604f0aaf190385.js?body=1 200 444.991 ms - -
GET /assets/components/auth-credentials.self-2b60839b945e22a923768c5e30ab8fed237bbbfe6d571d20998829f0bf704250.js?body=1 200 372.020 ms - -
GET /assets/components/cloud_volume_backup/cloud-volume-backup-form.self-f52fb93eb9649c12565a77881540671233499f650a25c7a0bcb29fd20da1254e.js?body=1 200 334.782 ms - -
GET /assets/components/datetime-delay-picker.self-b0b16402260f57d802b7680d8d1eb59aa87da750dd6ee91ac070d99e70c36482.js?body=1 200 333.968 ms - -
GET /assets/components/generic_object/assign-buttons.self-9bf233bce15b5c4fb24abe29cd4ba357654dffa9f66b6bb3384be783731014ce.js?body=1 200 334.152 ms - -
GET /assets/components/generic_object/custom-button-group-form.self-07182670a2819b0664130ce9a50b48389fb84f31d4431fe568f683758612cdd1.js?body=1 200 332.720 ms - -
GET /assets/components/generic_object/custom-image.self-2173d6c223ebb24594124c94a821206086bb07f6e20a80d9ffbf985ebe52d7c5.js?body=1 200 334.307 ms - -
GET /assets/components/generic_object/generic-object-definition-toolbar.self-704ed774b6ee742d3c8af33bc5c8d3746537b58e2dbca5653be26fd37a512b26.js?body=1 200 341.017 ms - -
GET /assets/components/generic_object/generic-object-definition.self-34b3380b889fc0a95856a106d3f5c5a30fee7b151195f62eba2a3d1ce4260795.js?body=1 200 347.102 ms - -
GET /assets/components/generic_object/generic-object-table.self-4ccdf32d0872a96d94285292a5e8b75abaa5959fa426d5497e026d0318f2cb98.js?body=1 200 347.489 ms - -
GET /assets/components/generic_object/main-custom-button-form.self-d8101a25df339d4e2235a1675961fc9391c7ceb934978eb19a1eef273171d338.js?body=1 200 343.472 ms - -
GET /assets/components/generic_object/main-custom-button-group-form.self-b288c82bf48548932269202043392b52df3e18959c050d9a4a01c3bccc391fd7.js?body=1 200 347.768 ms - -
GET /assets/components/miq-button.self-56d351d7138082a3e3b4291d484eaf941584eb2b543e5d9442087d804285d2b5.js?body=1 200 335.706 ms - -
GET /assets/components/miq-request-options.self-ca182f90423e9470bdace77ef8c774a5c117270e5556a321910f68c1c6aaf9c7.js?body=1 200 334.882 ms - -
GET /assets/components/miq_ae_class/namespace-form.self-a7b24a69ae42620f7e8f9f3fdf48b879df1cb3784f39a4678d4c4824bf389afb.js?body=1 200 333.312 ms - -
GET /assets/components/ops/log-collection-form.self-b85a1dd4b74266286ddf06e06cae7d4bdbb1239a75818dbeb02d5e466ee0f00f.js?body=1 200 332.153 ms - -
GET /assets/components/pf_charts/aggregate-status-card.component.self-5a9cb80b130e9f6f14d8c477fcc12e5aace0a3d5638e1446dbb559b62a561b06.js?body=1 200 329.838 ms - -
GET /assets/components/pf_charts/c3-chart.component.self-56d15e07537b2e2a64c4a7da5817abb7bb73c48189dbaf9d9e451d00b9060f20.js?body=1 200 332.453 ms - -
GET /assets/components/pf_charts/donut-chart.component.self-bda3416a4e68e5703caefbed07817b02f61a41ecf77d44d685fac00ec077a9ef.js?body=1 200 330.780 ms - -
GET /assets/components/pf_charts/donut-pct-chart.component.self-2db047ba324a9b09205ff65ea85ef9975ac34815561a6a77be3cae7dd9cc1cb9.js?body=1 200 332.060 ms - -
GET /assets/components/pf_charts/empty-chart.component.self-2335b761f31f6eb192669f8c23db2bae1e0f914618b44568b09f0a4317871da0.js?body=1 200 332.706 ms - -
GET /assets/components/pf_charts/group-chart.component.self-81b7938003a133995c443878b7c139b93437228fd9ab44f206c43e0a5c1470cf.js?body=1 200 331.894 ms - -
GET /assets/components/pf_charts/heatmap-legend.component.self-69ae79bb9fdd74dee4bec39a2569db2c8c9db6f3003c07f37633334834e4f4e9.js?body=1 200 335.799 ms - -
GET /assets/components/pf_charts/heatmap.component.self-d1dea8cbb21ee6ead99d45d9c2dd404a4c14c4b7643a7a4d4fe97ab4f20ccc9e.js?body=1 200 336.213 ms - -
GET /assets/components/pf_charts/line-chart.component.self-e95aa66a6b0d9742c76d680d6fd3c54f98b470d1aea0c27f9b4df41105f7daf8.js?body=1 200 334.738 ms - -
GET /assets/components/pf_charts/pie-chart.component.self-d75300f79c8eec8da57e6675a6ff8ae883330c4ad1f0dc0230246f3f79d521e7.js?body=1 200 332.810 ms - -
GET /assets/components/pf_charts/sparkline-chart.component.self-990b45be4f691c7d143bb5cceede38a0d1571f74ee02fdec6f2a1232a75f696a.js?body=1 200 326.610 ms - -
GET /assets/components/pf_charts/trends-chart.component.self-028f0d62ccb5f7dc817649f69b074f506ca3d17d83954399314c0b318b25df64.js?body=1 200 328.489 ms - -
GET /assets/components/pf_charts/utilization-trend-chart.component.self-c07c01ff35b1dc31bcc49409f64f6eaf666242037999d31d4312233d81530e0f.js?body=1 200 329.659 ms - -
GET /assets/components/physical_infrastructures/change-password.self-5f6f7c60d4380d9919a3b742eb3687bdf53e95806b7d7098fb46753ed6b9ebbc.js?body=1 200 335.537 ms - -
GET /assets/components/provider-option-field-input.self-9a0f2b0916ced7a716e7f7fe77b2c7928c8190dcbb4aaac3248848fbd2eb938b.js?body=1 200 332.708 ms - -
GET /assets/components/provier-option-section.self-e45fde6a23bc82560dadb567dc4b0c646b1a485cda10bffceb0e03d0189c2b64.js?body=1 200 338.606 ms - -
GET /assets/components/recent-resource.self-14a65230a3c53ffa59c363193c6d5e91dd2a72d441d55ee72dfe564e4251ddc6.js?body=1 200 443.322 ms - -
GET /assets/components/retirement-form.self-e6313a8a4bf498c88c645ff597be1c7d00e99af9d14b5ccd387a3dca5a288188.js?body=1 200 444.076 ms - -
GET /assets/components/sanitize.self-3cbf499eaceaf45958f9487e04e11c4b383e7446e44addb429734ec3f7f25bb5.js?body=1 200 441.086 ms - -
GET /assets/components/verify-button.self-7201c86f79c09d9d54d8bda458fef5165b66b3503ae430a54de81d83f36a6b9f.js?body=1 200 438.552 ms - -
GET /assets/components/vm_cloud/vm-cloud-add-security-group.self-6561ad114405d914e1514465cbc23f5a2efe7dd2518b3fefe42800efb4877490.js?body=1 200 430.455 ms - -
GET /assets/components/vm_cloud/vm-cloud-remove-security-group.self-7000035cefd484d269fe883b847ca41da06d84255a1eac25c15b32fab2f4169f.js?body=1 200 340.125 ms - -
GET /assets/components/vm_cloud/vm_cloud_evacuate_form.self-4940f31c3002fe24722a74afe45b7048a173286681deb536a1f28b3e568114cc.js?body=1 200 333.659 ms - -
GET /assets/components/cloud_subnet/cloud-subnet-form.self-a1951ca6801497dcb6a6bc82c20be37e2c468dc2bbefd3b3a167aca3a4a5da52.js?body=1 200 3048.418 ms - -
GET /assets/services/bootstrap_tree_subscription_service.self-595d93c2776554f5148c7a695202ccbcaa6e00ddefa2d3cc8fdcec7417e60b14.js?body=1 200 329.971 ms - -
GET /assets/services/catalog/catalog_item_data.self-dd75df5bc18fe156f5a5f5c2fed9bccb205acfd0497347e5b1c99e4db28b3680.js?body=1 200 335.246 ms - -
GET /assets/services/configuration_manager_service/configuration_manager_service.self-ac6c263498ee0e52fd76ffdb5a9bef819ed1da0909c1a1ca3cc86bd32a443f43.js?body=1 200 331.319 ms - -
GET /assets/services/dashboard_service.self-41112d805be807d8345d962f2215ff488a2e153d047ea9c76dcaf147010a8eee.js?body=1 200 336.133 ms - -
GET /assets/services/dialog_editor_http_service.self-ffda05a3c445bd3b56708a6d18bd5f893e193a402869ea306081b84ddd5beda2.js?body=1 200 334.986 ms - -
GET /assets/services/dialog_field_refresh_service.self-b6fedc8c23abbe91cf7b108eff5ae3c8d9e2bf1ffa56a6268bbb9ece1aff9940.js?body=1 200 344.845 ms - -
GET /assets/services/dialog_user_submit_error_handler_service.self-65fb51f5916fcffd8fd7fb628f365b2b7ecf9134eb08125cae916fc3511c4dd9.js?body=1 200 337.008 ms - -
GET /assets/services/alerts_center_service.self-e15c7c203c20bb5485358a152e170bef7fc0359ac5c1014abfdd77c58642ee4d.js?body=1 200 931.483 ms - -
GET /assets/services/miq_db_backup_service.self-08fbbecf2f3c3e337973b46e980829a843adfa34dd094c6217f5b1a7c64b4d7b.js?body=1 200 345.491 ms - -
GET /assets/services/post_service.self-0150eebc7aacc1045e4410dc8b5ed5154e226359f6118d4ebf7996b67f300a5f.js?body=1 200 345.590 ms - -
GET /assets/services/miq_service.self-e3115a57cdb93d7e5b8b0af48cba2369a0e00e624f678e8d4fb856d82063e33a.js?body=1 200 454.757 ms - -
GET /assets/services/subscription_service.self-b4bda6f6f0e63c405a4960ca10fd7d721b798b0e15b6c43bd5f76821673af98f.js?body=1 200 395.401 ms - -
GET /assets/services/topology_service.self-c9379e5c0bca20dadde0f9c0055217468910c9a911a16dfbb5ca868251cf3a91.js?body=1 200 389.782 ms - -
GET /assets/miq_browser_detect.self-fbf48f44e8dce67fa281c167defd6b9fbf80033187163a136102e7d55eb056e1.js?body=1 200 387.070 ms - -
GET /assets/miq_application.self-aa6a983f12d9ca619079158a1e05a4db63de9e8554eebf6ee77e5e29db9e16cc.js?body=1 200 386.110 ms - -
GET /assets/miq_flash.self-efe218af5e71bbdec061db991f89187bb8a2a6202c04703f3a9d0f802fec720a.js?body=1 200 382.522 ms - -
GET /assets/miq_change_stored_password.self-b1a67b98d69c9a42013dd74c6a7040da4fbff2e799dbf3906453e149c3ca357a.js?body=1 200 389.899 ms - -
GET /assets/miq_qe.self-02fe8d926bce59ad0b4f74ac8aa489273fb3fe85f7f6299554b43184fd531fd8.js?body=1 200 357.100 ms - -
GET /assets/git_import.self-838d28e444821b51d0bd5149aed04ef0d25b9c127b0e4669bb0088d5da4c409d.js?body=1 200 356.127 ms - -
GET /assets/import.self-ff7a5752228a165c5e748f0d58b59a46dcd740bad19206f48d2d4b4c6d825b61.js?body=1 200 348.734 ms - -
GET /assets/automate_import_export.self-b68d3d73c99ac0386907fecdbc76a16c1894a3303f068818bcea9ef102b9029a.js?body=1 200 346.464 ms - -
GET /assets/dialog_field_refresh.self-7e891ec3494a3edaef1423232c48ffc1a9625b8c6b3f8169367f8335f9106c44.js?body=1 200 342.518 ms - -
GET /assets/miq_c3_config.self-a2c0cc3d7bb34d511a0330a07346adfa2580a8d31f6d1cdaa9ffad2577099db3.js?body=1 200 339.151 ms - -
GET /assets/miq_ujs_bindings.self-18ba03372e98322ad75b533b8a39ed8e5378b56201107f5b005f9789c6ef0833.js?body=1 200 337.470 ms - -
GET /assets/miq_tree.self-2027316679f7453cf6dcd57757086c51ab4720a9ffd78cd26ba1094df39446b5.js?body=1 200 338.452 ms - -
GET /assets/miq_formatters.self-ce4b8e42c62955443827c716d1072357a04813393bcd6af7f5a0445144012fc1.js?body=1 200 336.218 ms - -
GET /assets/miq_grid.self-4d535ec7c9e48fc00a6d811493fbff1f0e4120b0ae34518d78f22f47feadfc8d.js?body=1 200 333.994 ms - -
GET /assets/miq_list_grid.self-f106940782020d75390736ca9d5118aefcfd7ec3c091e7b96ea8f5dd089315a7.js?body=1 200 344.469 ms - -
GET /assets/miq_toolbar.self-1dd06d544767dc9974df95b51ee5fef7a487e9ad3a7b263019dffdc5c8e695be.js?body=1 200 340.113 ms - -
GET /packs/manageiq-ui-classic/application-common-b9bf17a5fb9cbec243c5.js 200 3.940 ms - -
GET /packs/manageiq-ui-classic/angular-common-8824e9177498da64d45c.js 200 3.772 ms - -
GET /assets/miq_c3.self-51ba170a4c6b0856bcea9d518cb01c261fb4463ba0480639097824b4577507ed.js?body=1 200 449.231 ms - -
GET /packs/manageiq-ui-classic/provider-dialogs-common-dec78427eff0d58f38b6.js 200 4.219 ms - -
GET /packs/manageiq-ui-classic/provisioning-actions-common-1fc10b00ecfa6585615d.js 200 2.979 ms - -
GET /packs/manageiq-ui-classic/toolbar-actions-common-16402a56f4a5b765479c.js 200 4.735 ms - -
GET /assets/miq_explorer.self-6c78e97c1d1cfdaf0c3705e6d1c2c95ccd265d68277209720d20108463e94db5.js?body=1 200 445.184 ms - -
GET /packs/manageiq-providers-lenovo/component-definitions-common-263d8a8005e0d4d25223.js 200 8.058 ms - -
GET /packs/manageiq-ui-classic/component-definitions-common-dd79b55792eee324d837.js 200 3.483 ms - -
GET /packs/manageiq-providers-nsxt/component-definitions-common-04b2ddc2d48b741bb40a.js 200 8.862 ms - -
GET /packs/manageiq-providers-nuage/component-definitions-common-ca09234f868df5fc6b89.js 200 10.039 ms - -
GET /assets/miq_timeline.self-5b70ba88fb96851415999cb2f7d79acbcd472b7713c89b17d3174a48deed9c90.js?body=1 200 468.010 ms - -
GET /packs/manageiq-providers-redfish/component-definition-common-53b9b1ea8b82afff6dfc.js 200 239.296 ms - -
GET /assets/application.self-9e2a078724c35cb29f9c1d6c2894d45b03cf53499867de8494fe8e6a0f52d1bf.js?body=1 200 505.108 ms - -
GET /assets/services/timer_option_service.self-6f29a11b576d27cfe9b760a997b51ef80e756f3d201835dc6846657d78eec98a.js?body=1 200 2175.881 ms - -
GET /assets/miq_global.self-b20da38d6bd2b161ff081db1b03a3a7771a6c5ec02fb7c18a0b781b5872e5557.js?body=1 200 14026.595 ms - -
GET /assets/miq_debug.self-fea17ab96fea3abe999cf2832b839a884e72ed6f9d862f8a058ca6c168596c51.css?body=1 200 306.951 ms - -
GET /assets/miq_debug.self-1351f771c2cd2fab0d83069fd3f62aadbb7effc11a01ed4942f52308793453be.js?body=1 200 706.749 ms - -
GET /assets/locale/es/app.self-f8b8c20d4824f56b78abb9d672a150a5827cd570229ebfdf5a71a580cd96c9a9.js?body=1 200 14380.302 ms - -
GET /assets/locale/pt_BR/app.self-6b444cb234bb9c5622237f813273043a4b203817ae18920ffce457af8423e699.js?body=1 200 14438.650 ms - -
GET /assets/locale/fr/app.self-d94cefd3cbb2393d7a5
```

</details>

Reply:
That's a rails issue, so, I guess there's more places to add CYPRESS_DEV logic to?

Reply:
(Or, if it's this trivial to use a different db, we can just run `DB=vmdb_integration rails s` in development and there's no issue, right?)

Reply:
> we can just run `DB=vmdb_integration rails s`...

FYI:  This isn't a default rails thing, and I have it set up in my `config/database.yml` to do the following:

```yaml
---
base: &base
  adapter: postgresql
  encoding: utf8
  username: root
  pool: 5
  wait_timeout: 5
  min_messages: warning

development:
  <<: *base
  database: <%= ENV['DB'] || "vmdb_development" %>
  min_messages: notice

integration:
  <<: *base
  database: <%= ENV['DB'] || "vmdb_integration" %>

production:
  <<: *base
  database: <%= ENV['DB'] || "vmdb_production" %
```

Allowing me to use `$DB` to define what database I am using.

> Or, if it's this trivial to use a different db...

`tl;dr`:  No, it isn't that simple.

The database is less of the issue, and more the environment we are running in.  Using the `integration` environment, we are going to expose routes that allow you to clean the database (`/api/integration/clean`), or seed records (`api/integration/factory`) that are specific to that environment and are not exposed anywhere else.  This will allow the `cypress` tests to have a "given state" prior to running a particular spec instead of having to, for example, login every time by filling out the form fields at the login page, but I don't think that is the only case we are referring to.

The whole point of this is to try and get us to a state where we can match a specific point of Cypress' best practices:

https://docs.cypress.io/guides/references/best-practices.html#Organizing-Tests-Logging-In-Controlling-State

And I am doing it in a way that doesn't require us to mangle other environments in the process.

Reply:
Oh, interesting idea to push this over the API, but why not, yeah :+1: 

But .. if that's so .. should this be..

```
ENV["NODE_ENV"] ||= RAILS_ENV == "integration" ? (CYPRESS_DEV ? "development" : "production") : RAILS_ENV
```

then? Because otherwise you get development-like asset pipeline and production-like webpack, right?